packages/grafana-ui/src/utils/useMousePosition.ts

```diff
@@ -0,0 +1,27 @@
+import { throttle } from 'lodash';
+import { useState, useEffect } from 'react';
+
+interface MousePosition {
+  x: number | null;
+  y: number | null;
+}
+
+// For performance reasons, we throttle the mouse position updates
+const DEFAULT_THROTTLE_INTERVAL_MS = 50;
+
+export const useMousePosition = (throttleInterval = DEFAULT_THROTTLE_INTERVAL_MS) => {
+  const [mousePosition, setMousePosition] = useState<MousePosition>({ x: null, y: null });
+
+  useEffect(() => {
+    const updateMousePosition = throttle((event: MouseEvent) => {
+      setMousePosition({ x: event.clientX, y: event.clientY });
+    }, throttleInterval);
+    window.addEventListener('mousemove', updateMousePosition);
+
+    return () => {
+      window.removeEventListener('mousemove', updateMousePosition);
+    };
+  }, [throttleInterval]);
```

Comment:
Have you tried instead using requestAnimationFrame as your throttling mechanism? You listen to every mousemove event, but only queue up one `setMousePosition` call per requestAnimationFrame?

Reply:
good shout on this 👍 very smooth 🚀